neo4j/src/main/java/org/apache/zeppelin/graph/neo4j/Neo4jCypherInterpreter.java

```diff
@@ -0,0 +1,275 @@
+import org.apache.zeppelin.interpreter.graph.GraphResult;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.neo4j.driver.internal.types.InternalTypeSystem;
+import org.neo4j.driver.internal.util.Iterables;
+import org.neo4j.driver.v1.Record;
+import org.neo4j.driver.v1.StatementResult;
+import org.neo4j.driver.v1.Value;
+import org.neo4j.driver.v1.types.Node;
+import org.neo4j.driver.v1.types.Relationship;
+import org.neo4j.driver.v1.types.TypeSystem;
+import org.neo4j.driver.v1.util.Pair;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Neo4j interpreter for Zeppelin.
+ */
+public class Neo4jCypherInterpreter extends Interpreter {
+  private static final String TABLE = "%table";
+  public static final String NEW_LINE = "\n";
+  public static final String TAB = "\t";
+
+  private static final String MAP_KEY_TEMPLATE = "%s.%s";
+
+  private Map<String, String> labels;
+
+  private Set<String> types;
+  
+  private final Neo4jConnectionManager neo4jConnectionManager;
+  
+  private final ObjectMapper jsonMapper = new ObjectMapper();
+
+  public Neo4jCypherInterpreter(Properties properties) {
+    super(properties);
+    this.neo4jConnectionManager = new Neo4jConnectionManager(properties);
+  }
+
+  @Override
+  public void open() {
+    this.neo4jConnectionManager.open();
+  }
+
+  @Override
+  public void close() {
+    this.neo4jConnectionManager.close();
+  }
+
+  public Map<String, String> getLabels(boolean refresh) {
+    if (labels == null || refresh) {
+      Map<String, String> old = labels == null ?
+          new LinkedHashMap<String, String>() : new LinkedHashMap<>(labels);
+      labels = new LinkedHashMap<>();
+      StatementResult result = this.neo4jConnectionManager.execute("CALL db.labels()");
+      Set<String> colors = new HashSet<>();
+      while (result.hasNext()) {
+        Record record = result.next();
+        String label = record.get("label").asString();
+        String color = old.get(label);
+        while (color == null || colors.contains(color)) {
+          color = Neo4jConversionUtils.getRandomLabelColor();
+        }
+        colors.add(color);
+        labels.put(label, color);
+      }
+    }
+    return labels;
+  }
+
+  private Set<String> getTypes(boolean refresh) {
+    if (types == null || refresh) {
+      types = new HashSet<>();
+      StatementResult result = this.neo4jConnectionManager.execute("CALL db.relationshipTypes()");
+      while (result.hasNext()) {
+        Record record = result.next();
+        types.add(record.get("relationshipType").asString());
+      }
+    }
+    return types;
+  }
+
+  @Override
+  public InterpreterResult interpret(String cypherQuery, InterpreterContext interpreterContext) {
+    logger.info("Opening session");
+    if (StringUtils.isEmpty(cypherQuery)) {
+      return new InterpreterResult(Code.ERROR, "Cypher query is Empty");
```

Comment:
could you check other interpeters? it might be preferred to do nothing if the query is empty.